OpenRA.Mods.Common/Traits/Production.cs

```diff
@@ -120,25 +120,62 @@ public virtual void DoProduction(Actor self, ActorInfo producee, ExitInfo exitin
 
 				var notifyOthers = self.World.ActorsWithTrait<INotifyOtherProduction>();
 				foreach (var notify in notifyOthers)
-					notify.Trait.UnitProducedByOther(notify.Actor, self, newUnit);
+					notify.Trait.UnitProducedByOther(notify.Actor, self, newUnit, type);
 
 				foreach (var t in newUnit.TraitsImplementing<INotifyBuildComplete>())
 					t.BuildingComplete(newUnit);
 			});
 		}
 
-		public virtual bool Produce(Actor self, ActorInfo producee, string factionVariant)
+		protected virtual ExitInfo SelectExit(Actor self, ActorInfo producee, string type, Func<ExitInfo, bool> p)
+		{
+			if (type != null)
+			{
+				var exit = self.RandomExitOrDefault(type, p);
+				if (exit != null)
+					return exit;
+			}
+
+			var bi = producee.TraitInfoOrDefault<BuildableInfo>();
+			if (bi != null)
+			{
+				if (Info.Produces.Contains(bi.BuildAtProductionType))
```

Comment:
IOW: I can see I can use `BuildAtProductionType` to define separate exits for actors on a `ClassicProductionQueue` with this PR, but I don't see the same applying to `ProductionQueue`s.